url2698a2c731_cassandra_project_cassandra_stand_alone_tgz-pJ8-eu_cassandra_sim_MySimpleDynamicScenarioSimulationJ8/src/eu/cassandra/sim/entities/appliances/ConsumptionModel.java

```diff
@@ -0,0 +1,358 @@
+
+		try {
+
+			try {
+				outerN = ((Integer)modelObj.get("n")).intValue();
+			} catch(ClassCastException e ) {
+				try {
+					outerN = ((Long)modelObj.get("n")).intValue();
+				} catch(ClassCastException e1 ) {
+					outerN = ((Double)modelObj.get("n")).intValue();
+				}
+			} catch(NullPointerException npe) {
+				throw new Exception("Bad Parameter Exception: outer iterations parameter name should be n");
+			}
+			BasicDBList patternsObj = (BasicDBList)modelObj.get("params");
+			patternN = patternsObj.size();
+			patterns = new ArrayList[patternN];
+			n = new int[patternN];
+			patternDuration = new int[patternN];
+			for(int i = 0; i < patternN; i++) {
+				try {
+					n[i] = ((Integer)((DBObject)patternsObj.get(i)).get("n")).intValue();
+				} catch(ClassCastException e ) {
+					try {
+						n[i] = ((Long)((DBObject)patternsObj.get(i)).get("n")).intValue();
+					} catch(ClassCastException e1 ) {
+						n[i] = ((Double)((DBObject)patternsObj.get(i)).get("n")).intValue();
+					}
+				} catch(NullPointerException npe) {
+					throw 
+					new Exception("Bad Parameter Exception: inner iterations parameter name should be n");
+				}
+				BasicDBList values = ((BasicDBList)((DBObject)patternsObj.get(i)).get("values"));
+				int tripplets = values.size();
+				patterns[i] = new ArrayList<Tripplet>(tripplets);
+				for(int j = 0; j < tripplets; j++) {
+					Tripplet t = new Tripplet();
+					try {
+						t.v = ((Double)((DBObject)values.get(j)).get(type)).doubleValue();
+					} catch(ClassCastException e) {
+						try {
+							t.v = ((Integer)((DBObject)values.get(j)).get(type)).intValue();
+						} catch(ClassCastException e1) {
+							t.v = ((Long)((DBObject)values.get(j)).get(type)).intValue();
+						}
+					} catch(NullPointerException npe) {
+						throw 
+						new Exception("Bad Parameter Exception: power parameter name should be " + type);
+					}
+					try {
+						t.d = ((Integer)((DBObject)values.get(j)).get("d")).intValue();
+					} catch(ClassCastException e) {
+						try {
+							t.d = ((Double)((DBObject)values.get(j)).get("d")).intValue();
+						} catch(ClassCastException e1) {
+							t.d = ((Long)((DBObject)values.get(j)).get("d")).intValue();
+						}
+					} catch(NullPointerException npe) {
+						throw 
+						new Exception("Bad Parameter Exception: duration parameter name should be d");
+					}
+					patternDuration[i] += t.d; 
+					totalDuration += (n[i] * t.d);
+					try {
+						t.s = ((Double)((DBObject)values.get(j)).get("s")).doubleValue();
+					} catch(ClassCastException e) {
+						try {
+							t.s = ((Integer)((DBObject)values.get(j)).get("s")).intValue();
+						} catch(ClassCastException e1) {
+							t.s = ((Long)((DBObject)values.get(j)).get("s")).intValue();
+						}
+					} catch(NullPointerException npe) {
+						throw 
+						new Exception("Bad Parameter Exception: slope parameter name should be s");
+					}
+					patterns[i].add(t);
+				}
+			}
+		} catch(Exception bpe) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)